cirq/protocols/channel.py

```diff
@@ -122,3 +123,46 @@ def channel(val: Any,
                         "_unitary_ method.".format(type(val)))
     raise TypeError("object of type '{}' does have a _channel_  or _unitary_ "
                     "method, but it returned NotImplemented.".format(type(val)))
+
+
+def stochastic_unitary(
```

Comment:
I think we should delegate this task to the gates. Gates will now if they are equivalent to a stochastic unitary, and they can tell us instead of us trying to figure out it after forcing the gate explain it in a different language.

Basically, we should have a `_stochastic_unitary_` magic method.

Reply:
That's not DRY.  By having this method this means that you only need to implement channel.  In what you propose you would have to write essentially the same operators twice. We can avoid this by just a small amount of math.

Reply:
That's very unDRY.  If you have this magic method you would essentially repeat the code for the channel multiple times with subtle differences.  This means writing essentially the same code twice, and worse it can lead to inconsistencies.  By using this you never run into that problem at the cost of a small, albeit very standard, bit of linear algebra.

Reply:
Oh, you would avoid repetition by modifying `cirq.channel` to fallback onto `_stochastic_unitary_` and then onto `_unitary_`.

That being said, repetition can be justified by performance benefits and this could fall under that umbrella.

Reply:
My first thought is that i think that would be confusing for people.  Knowing that you could implement _stochastic_unitary_ would be something you would have to discover.  (And I bet most people would implement both _channel_ and _stochastic_unitary_).  I guess stochastic_unitary could fall back to attempting to decompose the channel.  Right now the only plans for this are that the simulator knows about it.  I do agree that performance might be worth thinking about.  The other problem is that there are other channel classes beyond stochastic unitary (unital for example) and I wonder where we draw the line.  

OK let me think about this a bit.